### PR TITLE
org.freedesktop.sdk.extension.openjdk17

### PR DESCRIPTION
**Pull request by mbooth101**
This the new LTS release of the JDK.

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
**I am not a contributor upstream, but the upstream Java team at Red Hat already know I maintain the other JDK extensions at Flathub**
- [ ] ~~I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).~~
**This is a SDK extension, not an application**
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
**Included patch was taken from Red Hat's JDK distro and they are the upstream maintainers.**

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Note I also am the maintainer of the other OpenJDK extensions in Flathub:
* https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk
* https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk11

### CODE REVIEW

**Comment by mbooth101** [comment]
Do I have permission to dispatch the bot?

bot, build org.freedesktop.sdk.extension.openjdk17

**Comment by mbooth101** [comment]
Errr, the test build cannot deal with yaml-flavour manifests? What's going on there?

**Comment by mbooth101** [comment]
bot, build org.freedesktop.sdk.extension.openjdk17

**Comment by mbooth101** [comment]
bot, build org.freedesktop.sdk.extension.openjdk17.yaml

**Comment by mbooth101** [comment]
@barthalion @nedrichards  Hi both, any idea why Mr Flathubbot cannot build my PR?

**Comment by mbooth101** [comment]
Oh wait, do I need to use perfect casing?

bot, build org.freedesktop.Sdk.Extension.openjdk17

File: org.freedesktop.Sdk.Extension.openjdk17.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          - x86_64
+        url: https://fedorapeople.org/~mbooth/bootstrap_jdk/bootstrap-openjdk-16.0.1.0.9-1.rolling.fc32.x86_64.tar.bz2
+        dest-filename: java-openjdk.tar.bz2
+        sha512: 0084f99558a73788438df4fe62700584b41ee962d7de7bd322d8950207c88acfb3839c788def906926325f782ae86068315352dfc3b8c6d5375dec192cf3adeb
+      - type: file
+        only-arches:
+          - arm
+        url: https://fedorapeople.org/~mbooth/bootstrap_jdk/bootstrap-openjdk-16.0.1.0.9-1.rolling.fc32.armv7hl.tar.bz2
+        dest-filename: java-openjdk.tar.bz2
+        sha512: 9fad9ef72174ad268ecf0d85b0eef8a6ee54c68de4060947a9bf2593c21a9cf0bc119c068cde67163b402cc2da972a79a7f1d7875d63d6899dcffd060689fa08
```
Comment (hfiguiere) [inline org.freedesktop.Sdk.Extension.openjdk17.yaml:36]: this is unecessary armv7 is no longer supported in either 20.08 or 21.08

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.freedesktop.Sdk.Extension.openjdk17.yaml (hunk 1 above)
Comment (mbooth101) [reply org.freedesktop.Sdk.Extension.openjdk17.yaml:36]: Yep, I note that 19.08 is now EOL :-)

**Comment by mbooth101** [comment]
Hmm, I am experiencing a internal compiler error in GCC that is preventing me building JDK17 successfully.

I filed a bug here: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1330

In the meantime, I'll try to work around it.

**Comment by barthalion** [comment]
FWIW there's org.freedesktop.Sdk.Extension.llvm12 available. Not sure if it fixes your build error, though.

**Comment by mbooth101** [comment]
bot, build org.freedesktop.Sdk.Extension.openjdk17

**Comment by mbooth101** [comment]
bot, build org.freedesktop.Sdk.Extension.openjdk17

**Comment by mbooth101** [comment]
Okay next problem, I'm not sure why it fails to build on aarch64: 

```
checking for gcc... /usr/bin/gcc
checking resolved symbolic links for CC... /usr/bin/aarch64-unknown-linux-gnu-gcc-11.2.0
configure: Using gcc C compiler version 11.2.0 [gcc (GCC) 11.2.0]
checking whether the C compiler works... no
configure: error: in `/run/build/java':
configure: error: C compiler cannot create executables
```

**Pull request closed without merging**